// Implement Histograms as protocol buffers
// Requires separate Binning as in the json implementation
syntax = "proto3";
package physt;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";

message Meta {
    string name = 1;
    repeated string axis_names = 2;
}

message BinLimits {
    double lower = 1;
    double upper = 2;
}

message Binnings {
    bool adaptive = 1;
    string binning_type = 2;
    repeated BinLimits bins = 3;
}

// For nD histograms, binnings should be repeated
message Histogram {
    string histogram_type = 1;
    repeated Binnings binnings = 2;    
    repeated double frequencies = 3;
    string dtype = 4;
    repeated double errors2 = 5;
    Meta meta = 6;
    repeated double missed = 7;
    bool missed_keep = 8;
    string physt_version = 9;
    string physt_compatible = 10;
}

// Summary will be a map
// histogram name and histogram
// Ability to merge Summary files
// Keys should be unique, see language docs
message Summary {
    map<string, Histogram> histograms = 1;
}


